python/tvm/contrib/cutlass/gen_conv2d.py

```diff
@@ -252,6 +252,8 @@ def select_op(
             lambda align: all([dim % align == 0 for dim in [IC, OC]]),
             use_3xtf32,
             profile_all_alignments,
+            # Use fp32 accumulation for wgrad to align with cuDNN
+            accumlator_dtype="float32" if conv_kind == ConvKind.Wgrad else out_dtype,
```

Comment:
Here, we force the accum dtype to be fp32 if wgrad.

Reply:
This makes lots of sense. I'm even wondering whether we should force accum dtype to be fp32 for all ConvKind instead of just wgrad.

Reply:
Always using fp32 accum for other dtype will probably bring perf regression (cuDNN also allows fp16 accumulation for fprop and dgrad). Ideally we should add `accumulation_dtype` to `Conv2dAttr` to guide that decision, I thought about doing that, but I realized that I have to change a lot of topi to take that into account. 

Also we need to discuss what the interface for `ToMixedPrecision` should be if we want to allow changing accumulation dtype, right now we cannot flexibly change even output dtype @AndrewZhaoLuo

Reply:
I see. Make sense.
We discussed about the accum dtype before when @AndrewZhaoLuo was working on the ToMixedPrecision pass, but just like you pointed out, this will involve lots of TOPI changes.

Reply:
Hmm yeah, so if I'm understanding correctly for conv2d_winograd we want to accumulate to fp32 but if it's not winograd we are ok with accumulating to fp16. 

`ToMixedPrecision` can configure accumulation and output dtypes for any call node but only using information from examining that node. I'm not sure implementation details like whether it's winograd can be transmitted here. 

I will say on relay level all we care about is type checking imo so just get the output_dtype correct. For example, accumulate all you like in fp32 but internally just make sure the output fits interface. Perhaps extraneous cast here is bad but maybe we can repair it further down in topi-tir level.

Reply:
yeah, I agree that we can let `out_dtype` in conv2d essentially act like the accumulation data type, and add an explicit cast op if `accum_dtype != out_dtype`. That's fine as far as `ToMixedPrecision` pass goes, but for cutlass BYOC, we need to additionally pattern match against the added `cast(fp32 -> fp16)` op to know that this conv2d is fp32 accum -> fp16 out. And for cuDNN which is not implemented as BYOC, this doesn't work because all it sees is fp32 accum -> fp32 out conv2d. And for cuDNN wgrad doesn't support that. 



> Hmm yeah, so if I'm understanding correctly for conv2d_winograd we want to accumulate to fp32 but if it's not winograd we are ok with accumulating to fp16.

@AndrewZhaoLuo Here wgrad means "conv2d gradient with respect to weight", not winograd :)